### ISSUE TITLE
ScrollUp does not happen for searchText but only scrollDown, how to find a text on top of the view?

### ISSUE THREAD

**Issue by renas** [RobotiumTech/robotium#245]
_From [pr...@anvaya.net](https://code.google.com/u/102624229117984655215/) on September 27, 2011 16:41:46_

What steps will reproduce the problem? 1. searchText does not scroll up to search the test
2. use searchText to find a text that is well below the current view
then use searchText again which is at the top of the view which can be seen only if scrolling upwards is done. 3. What is the expected output? What do you see instead? Expect scrollUp() / scrollDown() to be performed until the text is found. ScrollUp() does not occur and the test fails unable to find the text at the top that is not visible on the current view but could be seen only by scrolling What version of the product are you using? On what operating system? solo2.2, Emulator with Google API 8. Please provide any additional information below.

_Original issue: http://code.google.com/p/robotium/issues/detail?id=166_

**Comment by renas** [comment]
_From [renasr...@gmail.com](https://code.google.com/u/108052454046732184536/) on September 27, 2011 23:21:54_

You will have to scroll up your self after using searchText the first time. scrollUp/scrollDown will give you false back when it is not able to scroll anymore. So you test case could look something like this:

solo.searchText();
while(solo.scrollUp());
solo.searchText();

**Status:** Invalid

**Comment by renas** [comment]
_From [pr...@anvaya.net](https://code.google.com/u/102624229117984655215/) on September 28, 2011 15:47:59_

solo.searchText();
while(solo.scrollUp());
solo.searchText();

i invalid. I tried this before I entered the issue. The reason it is invalid is because:
while loop scrolls up first.
Then searchtext makes the scroll down automatically.
So basically the same view is been checked endless between scroll up and search text. ScrollUp goes up one view, while searchtext brings down one view searching for text and while loop continues endlessly. This solution did not work.
